mesheryctl/internal/cli/root/pattern/export.go

```diff
@@ -0,0 +1,209 @@
+package pattern
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshery/server/handlers"
+	"github.com/layer5io/meshery/server/models"
+	"github.com/layer5io/meshkit/models/oci"
+	location "github.com/layer5io/meshkit/utils"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var (
+	save string
+	wd string
+)
+
+var exportCmd = &cobra.Command{
+	Use:   "export",
+	Short: "Export pattern",
+	Long:  `Export pattern to OCI format`,
+	Args:  cobra.MinimumNArgs(1),
+	Example: `
+// export a pattern to default path
+mesheryctl pattern export [pattern-name | ID]
+mesheryctl pattern export "my meshery design"
+
+// export pattern to the path that given by user
+mesheryctl pattern export [pattern-name | ID] -o [path]
+	`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return err
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			return err
+		}
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			utils.Log.Error(err)
+			return nil
+		}
+		pattern := ""
+		isID := false
+
+		if len(args) > 0 {
+			pattern, isID, err = utils.ValidId(mctlCfg.GetBaseMesheryURL(), args[0], "pattern")
+			if err != nil {
+				utils.Log.Error(ErrPatternInvalidNameOrID(err))
+				return nil
+			}
+		}
+
+		urlString := mctlCfg.GetBaseMesheryURL()
+		if isID {
+			// if pattern is a valid uuid, then directly fetch the pattern
+			urlString += "/api/pattern/" + pattern
+		} else {
```

Comment:
Trim the argument. 

What else should we be checking for here? Be sure to be practicing secure coding.

Reply:
We checking the id is valid or not ,if it is valid, then searching the pattern by its id else it searching for the pattern by its name. there is different api for id and name.